src/main/java/com/browserstack/automate/ci/jenkins/BrowserStackReportForBuild.java

```diff
@@ -0,0 +1,217 @@
+        }
+    }
+
+    public boolean generateBrowserStackReport() {
+        if (this.result.size() == 0) {
+            this.result.addAll(generateSessionsCollection(this.browserStackSessions));
+
+            if (this.result.size() > 0) {
+                this.result.sort(new SessionsSortingComparator());
+                this.resultMeta.addAll(this.result.subList(0, Math.min(this.result.size(), RESULT_META_MAX_SIZE)));
+                generateAggregationInfo();
+                return true;
+            }
+            return false;
+        }
+        return true;
+    }
+
+    private Build fetchBrowserStackBuild(@Nonnull BrowserStackClient client, @Nonnull String buildName) {
+        Build build = null;
+        try {
+            build = client.getBuildByName(buildName);
+        } catch (BuildNotFound bnfException) {
+            log(this.logger, "No build found by name: " + buildName);
+        } catch (BrowserStackException bstackException) {
+            log(this.logger, "BrowserStackException occurred while fetching build: " + bstackException.toString());
+        }
+
+        return build;
+    }
+
+    private List<Session> fetchBrowserStackSessions(@Nonnull BrowserStackClient client, @Nonnull String buildId) {
+        final List<Session> browserStackSessions = new ArrayList<Session>();
+        try {
+            browserStackSessions.addAll(client.getSessions(buildId));
+        } catch (BuildNotFound bnfException) {
+            log(this.logger, "No build found while fetching sessions for the buildId: " + buildId);
+        } catch (BrowserStackException bstackException) {
+            log(this.logger, "BrowserStackException occurred while fetching sessions: " + bstackException.toString());
+        }
+
+        return browserStackSessions;
+    }
+
+    private List<JSONObject> generateSessionsCollection(List<Session> browserStackSessions) {
+        return browserStackSessions.stream().map(this::convertSessionToJsonObject).collect(Collectors.toList());
+    }
+
+    private JSONObject convertSessionToJsonObject(Session session) {
+        final JSONObject sessionJSON = new JSONObject();
+
+        if (session.getName() == null || session.getName().isEmpty()) {
+            sessionJSON.put("name", session.getId());
+        } else {
+            sessionJSON.put("name", session.getName());
+        }
+
+        if (session.getDevice() == null || session.getDevice().isEmpty()) {
+            sessionJSON.put("browser", session.getBrowser());
+        } else {
+            sessionJSON.put("browser", session.getDevice());
+        }
+        sessionJSON.put("os", session.getOs());
+        sessionJSON.put("osVersion", session.getOsVersion());
+        sessionJSON.put("status", session.getBrowserStackStatus());
+
+        if (session.getBrowserStackStatus().equals(session.getStatus())) {
+            sessionJSON.put("userMarked", "UNMARKED");
```

Comment:
Are keeping this in upcase? Like being consistent with status `failed`, `passed`

Reply:
will make it consistent...